### PR TITLE
Deployment check taint

### PR DESCRIPTION
**Pull request by jdowni000**
Included env variables and a new function to allow the ability to schedule the checker pods on dedicated nodes. Also modified the code to still take in blank values for those env variables to schedule anywhere as it does today. This has been tested in lab for both dedicated nodes as well as blank values which successfully schedules correctly for both circumstances.

### CODE REVIEW

File: cmd/deployment-check/deployment.go [hunk 1]
```diff
@@ -31,6 +32,11 @@ import (
 	watchpkg "k8s.io/apimachinery/pkg/watch"
 )
 
+var (
+	// Toleration value to be set in deployment pod
+	tolerationValue = os.Getenv("TOLERATION_VALUE")
+)
+
```
Comment (jonnydawg) [inline cmd/deployment-check/deployment.go:39]: Put this var declaration in the main.go

File: cmd/deployment-check/Makefile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 IMAGE="kuberhealthy/deployment-check"
-TAG="v1.6.2"
+TAG="v1.7.2"
```
Comment (jonnydawg) [inline cmd/deployment-check/Makefile:2]: bumping a minor value should reset the patch to 0 (i.e v1.7.0)

File: cmd/deployment-check/deployment.go [hunk 3]
```diff
@@ -251,6 +264,30 @@ func createDeployment(ctx context.Context, deploymentConfig *v1.Deployment) chan
...
 }
 
+// createContainerConfig creates a container resource spec and returns it.
+func createToleration(t string) corev1.Toleration {
+
+	// return blank tolerations if no value provided in yaml file
+	if len(t) == 0 {
+		tolerations := corev1.Toleration{
+			Key:      "",
+			Operator: "Exists",
```
Comment (jonnydawg) [inline cmd/deployment-check/deployment.go:274]: use the constant `corev1.NodeSelectorOpExists` here

File: cmd/deployment-check/deployment.go [hunk 4]
```diff
@@ -251,6 +264,30 @@ func createDeployment(ctx context.Context, deploymentConfig *v1.Deployment) chan
...
+			Value:    "",
+			Effect:   "",
```
Comment (jonnydawg) [inline cmd/deployment-check/deployment.go:276]: Go ahead and leave the `Key` `Value` and `Effect` fields out if you are going to define them as blank -- they are optional and already [have defaults](https://pkg.go.dev/k8s.io/api/core/v1#Toleration).

File: cmd/deployment-check/deployment.go (hunk 1 above)
Comment (jdowni000) [reply cmd/deployment-check/deployment.go:39]: done

File: cmd/deployment-check/Makefile (hunk 2 above)
Comment (jdowni000) [reply cmd/deployment-check/Makefile:2]: done

File: cmd/deployment-check/deployment.go [hunk 5]
```diff
@@ -251,6 +264,30 @@ func createDeployment(ctx context.Context, deploymentConfig *v1.Deployment) chan
...
+		}
+		return tolerations
+	}
+
+	// Create toleration Key
+	tolerations := corev1.Toleration{
+		Key:      t,
+		Operator: "Equal",
+		Value:    "kuberhealthy",
+		Effect:   "NoSchedule",
```
Comment (jonnydawg) [inline cmd/deployment-check/deployment.go:286]: same with the constant `corev1.TaintEffectNoSchedule` here for `NoSchedule`

File: cmd/deployment-check/deployment.go [hunk 6]
```diff
@@ -251,6 +264,30 @@ func createDeployment(ctx context.Context, deploymentConfig *v1.Deployment) chan
...
+			Value:    "",
+			Effect:   "",
+		}
+		return tolerations
+	}
+
+	// Create toleration Key
+	tolerations := corev1.Toleration{
+		Key:      t,
+		Operator: "Equal",
```
Comment (jonnydawg) [inline cmd/deployment-check/deployment.go:284]: `corev1.TolerationOpEqual`

File: cmd/deployment-check/deployment.go (hunk 4 above)
Comment (jdowni000) [reply cmd/deployment-check/deployment.go:276]: done

File: cmd/deployment-check/deployment.go [hunk 7]
```diff
@@ -251,6 +264,30 @@ func createDeployment(ctx context.Context, deploymentConfig *v1.Deployment) chan
 	return createChan
 }
 
+// createContainerConfig creates a container resource spec and returns it.
```
Comment (jonnydawg) [inline cmd/deployment-check/deployment.go:267]: Update your comments please!!

File: cmd/deployment-check/deployment-check.yaml [hunk 8]
```diff
@@ -10,13 +10,17 @@ spec:
...
     - name: deployment
-      image: kuberhealthy/deployment-check:v1.6.2
+      image: kuberhealthy/deployment-check:v1.7.2
       imagePullPolicy: IfNotPresent
       env:
         - name: CHECK_DEPLOYMENT_REPLICAS
           value: "4"
         - name: CHECK_DEPLOYMENT_ROLLING_UPDATE
           value: "true"
+        - name: TOLERATION_VALUE
```
Comment (jonnydawg) [inline cmd/deployment-check/deployment-check.yaml:20]: Having this in the example assumes that everyone has these taints & node labels -- maybe put them under a new section in the readme?

**Review by jonnydawg — CHANGES_REQUESTED** [review]

File: cmd/deployment-check/deployment.go (hunk 3 above)
Comment (jdowni000) [reply cmd/deployment-check/deployment.go:274]: this changes it to an operator which breaks the code. I will use the above comment and just keep Operator

File: cmd/deployment-check/deployment.go (hunk 5 above)
Comment (jdowni000) [reply cmd/deployment-check/deployment.go:286]: oh i see now....yes ignore above comment. will do

File: cmd/deployment-check/deployment.go (hunk 6 above)
Comment (jdowni000) [reply cmd/deployment-check/deployment.go:284]: done

File: cmd/deployment-check/deployment.go (hunk 7 above)
Comment (jdowni000) [reply cmd/deployment-check/deployment.go]: done

File: cmd/deployment-check/deployment-check.yaml (hunk 8 above)
Comment (jdowni000) [reply cmd/deployment-check/deployment-check.yaml:20]: removed values for these and left as blank which should be default.

File: .ci/values.yaml [hunk 9]
```diff
@@ -71,10 +71,12 @@ check:
     timeout: 15m
     image:
       repository: kuberhealthy/deployment-check
-      tag: v1.6.2
+      tag: v1.7.0
     extraEnvs:
       CHECK_DEPLOYMENT_REPLICAS: "4"
       CHECK_DEPLOYMENT_ROLLING_UPDATE: "true"
+      TOLERATION_VALUE: ""
```
Comment (jonnydawg) [inline .ci/values.yaml:78]: No need to add blank env vars

Comment (jdowni000) [reply .ci/values.yaml]: I will remove these

**Review by jonnydawg — APPROVED** [review]

**Pull request merged**